README.md

```diff
@@ -280,6 +280,53 @@ change to a direct dependency. This process applies to development dependencies
 If you wish to re-pin all the dependencies, run `make refresh_all_requirements`.  It is advisable to do a full
 test-deploy-test cycle after this (the test after the deploy is required to test the lambdas).
 
+#### Logging conventions
+
+1.  Always use a module-level logger, call it `logger` and initialize it as follows:
+
+    ```python
+    logger = logging.getLogger(__name__)
+    ```
+
+2.  Do not configure logging at module scope. It should be possible to import any module without side-effects on 
+    logging. The `dss.logging` module contains functions that configure logging for this application, its Lambda 
+    functions and unit tests.
+    
+3.  When logging a message, pass either
+ 
+    * an f-string as the first and only positional argument or
+    
+    * a %-string as the first argument and substitution values as subsequent argumentss. Do not mix the two string 
+      interpolation methods. If you mix them, any percent sign in a substituted value will raise an exception.
```

Comment:
The issues i found with this was if the f-string evaluated to something containing %s, %d, ect.. the logger would complain that it expected more parameters. For example `logger.warning(f"hello {x}", "%s")` would cause an exception expecting more arguments. Has this case been handled?

Reply:
Yes, it is handled by disallowing this particular case. You shouldn't pass an f"" string as the first argument and then pass additional parameters. More details on that can be found here:

https://github.com/HumanCellAtlas/data-store/issues/829#issuecomment-358455378

Reply:
Sorry bad example. This is what i meant,
```python
import logging
x = '%s'
logging.warning(f"{x}")
```
but i tested it and it works out.